### PR TITLE
[RLlib] DD-PPO training iteration fn

### PR DESCRIPTION
**Pull request by sven1977**
DD-PPO training iteration fn implementation:
* Add Pendulum learning test for this algo to CI
* Atari comparative benchmarks (vs execution_plan version) pending ...

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/agents/ppo/ddppo.py [hunk 1]
```diff
@@ -186,6 +202,122 @@ def validate_config(self, config):
...
+    @override(PPOTrainer)
+    def training_iteration(self) -> ResultDict:
+        # Shortcut.
+        first_worker = self.workers.remote_workers()[0]
+
+        # Run sampling and update steps on each worker in asynchronous fashion.
+        sample_and_update_results = asynchronous_parallel_requests(
+            remote_requests_in_flight=self.remote_requests_in_flight,
+            actors=self.workers.remote_workers(),
+            ray_wait_timeout_s=1000.0,  # 0.0
```
Comment (avnishn) [inline rllib/agents/ppo/ddppo.py:239]: huh?

File: rllib/tuned_examples/ppo/pendulum-ddppo.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+pendulum-ddppo:
+    env: Pendulum-v1
+    run: DDPPO
+    stop:
+        episode_reward_mean: -300
```
Comment (avnishn) [inline rllib/tuned_examples/ppo/pendulum-ddppo.yaml:5]: debugging?

File: rllib/BUILD [hunk 3]
```diff
@@ -249,6 +249,16 @@ py_test(
     args = ["--yaml-dir=tuned_examples/ppo"]
 )
 
+py_test(
```
Comment (sven1977) [inline rllib/BUILD:252]: This task is now working properly. Due to proper hyperparam tuning.

File: rllib/agents/ppo/ddppo.py (hunk 1 above)
Comment (sven1977) [reply rllib/agents/ppo/ddppo.py:239]: Ah, sorry, was just trying some stuff. Basically, this makes it synchronous :) Will revert. ...

File: rllib/tuned_examples/ppo/pendulum-ddppo.yaml (hunk 2 above)
Comment (sven1977) [reply rllib/tuned_examples/ppo/pendulum-ddppo.yaml:5]: Which part?
* reward: It's able to get to -300.
* timesteps: Yeah, it does need up to 1M sometimes. DD-PPO is not a very stable algo, it seems. Especially on cont. action tasks. But even on Atari I have yet to find a good choice of hyperparams.

File: rllib/agents/ppo/ddppo.py (hunk 1 above)
Comment (sven1977) [reply rllib/agents/ppo/ddppo.py:239]: done

File: rllib/tuned_examples/ppo/pendulum-ddppo.yaml (hunk 2 above)
Comment (avnishn) [reply rllib/tuned_examples/ppo/pendulum-ddppo.yaml:5]: Oh for some reason I read this as CartPole and not Pendulum, my b

**Review by avnishn — APPROVED** [review]
This looks p much good to me. 

I think that we can find good ATARI hparams, but we'll probably need some more logging infos (for stddev and entropy in the case of PPO) and then we'll be able to get a sufficiently working DDPPO agent.

File: rllib/agents/ppo/ddppo.py (hunk 1 above)
Comment (sven1977) [reply rllib/agents/ppo/ddppo.py]: Huh? Ok, now it's fixed ... Forgot to push.

File: rllib/tuned_examples/ppo/pendulum-ddppo.yaml (hunk 2 above)
Comment (sven1977) [reply rllib/tuned_examples/ppo/pendulum-ddppo.yaml:5]: Haha, yeah -300 would be pretty bad for CartPole :)

File: rllib/BUILD (hunk 3 above)
Comment (sven1977) [reply rllib/BUILD:262]: It was pretty hard, actually.
It's good to start with just one worker and using the exact same hparams as the respective PPO version. Then increase num_workers and at the same time carefully adjust:
```
rollout_fragment_length
num_envs_per_worker
sgd_minibatch_size
num_sgd_iter
```

in short, anything that affects (per-worker) batch-size, and the time each worker spends on a decentralized update.
